lib/SILOptimizer/Transforms/SSADestroyHoisting.cpp

```diff
@@ -786,8 +786,10 @@ bool HoistDestroys::checkFoldingBarrier(
   SILValue address;
   if (auto *load = dyn_cast<LoadInst>(instruction)) {
     auto loadee = load->getOperand();
-    auto accessPath = AccessPath::compute(loadee);
-    if (accessPath.getStorage().hasIdenticalStorage(storage)) {
+    auto relativeAccessStorage = RelativeAccessStorageWithBase::compute(loadee);
+    if (relativeAccessStorage.getStorage().hasIdenticalStorage(storage)) {
+      if (relativeAccessStorage.cast == AccessStorageCast::Type)
```

Comment:
> [SSADestroyHoisting] Bitcasts obstruct folding.

This commit looks good!

> if (relativeAccessStorage.cast == AccessStorageCast::Type)

Maybe it's worth a short comment here: A type cast via unchecked_addr_cast or pointer_to_address may theoretically result in a value that is not layout equivalent to the original (trailing fields may be dropped). Therefore, the `load [take]` would not be semantically equivalent to the original `destroy_addr`

Reply:
Definitely!  Added a couple comments.